patroni/dcs/etcd.py

```diff
@@ -297,14 +297,20 @@ def _load_machines_cache(self):
 
 
 def catch_etcd_errors(func):
-    def wrapper(*args, **kwargs):
+    def wrapper(self, *args, **kwargs):
         try:
-            return func(*args, **kwargs) is not None
+            return func(self, *args, **kwargs) is not None
         except (RetryFailedError, etcd.EtcdException):
             return False
```

Comment:
Not sure. This branch doesn't log the error, we might want to still see the stack trace. A better question would be, should this log errors? I think it should, will take a look what the consequences are.